File: Python3/프로그래머스/2/42587. 프로세스/프로세스.py
def solution(priorities, location):
    process = [key for key in range(len(priorities))]
    
    locations = [0 for _ in range(len(priorities))]
    
    stack = []
    sequence = 1
    while process:
        value = process.pop(0)
        
        skip = False
        for other in process:
            if priorities[value] < priorities[other]:
                process.append(value)
                skip = True
                break
        
        if skip:
            continue
        
        locations[value] = sequence
        sequence += 1
    
    return locations[location]
        
        
        
        
    
    
    
    
    
    
    
    
    
    
    
    
    
#     answer = [0] * len(priorities)
#     sequence = 1
#     queue = [i for i in range(len(priorities))]
    
#     while queue:
#         result = 0
#         p = queue.pop(0)
#         for item in queue:
#             if priorities[item] > priorities[p]:
#                 queue.append(p)
#                 result = 1
#                 break
#         if result == 0:
#             answer[p] = sequence
#             sequence += 1
        
#     return answer[location]

File: C++/프로그래머스/2/42587. 프로세스/프로세스.cpp
#include <vector>
#include <queue>
#include <algorithm>

using namespace std;

queue<int> stack;
vector<int> sequence;

int solution(vector<int> priorities, int location) {
    int size = priorities.size();
    sequence.resize(size);
    
    for (int i = 0; i < size; i++) {
        stack.push(i);
    }
    
    int seq = 1;
    
    while (!stack.empty()) {
        int v = stack.front();
        
        if (priorities[v] == *max_element(priorities.begin(), priorities.end())) {
            //priorities가 가장 높을 경우
            stack.pop(); //스택에서 삭제
            priorities[v] = 0;
            sequence[v] = seq;
            seq++;
            
        } else {
            stack.pop();
            stack.push(v); //다시 넣기
        }
    }
    
    return sequence[location];
}